src/Components/CreateImageWizard/steps/Oscap/components/PolicySelector.tsx

```diff
@@ -0,0 +1,314 @@
+    for (const pkg of oscapPackages) {
+      dispatch(removePackage(pkg));
+    }
+  };
+
+  const handlePartitions = (oscapPartitions: Filesystem[]) => {
+    dispatch(clearPartitions());
+
+    const newPartitions = oscapPartitions.map((filesystem) => {
+      const [size, unit] = parseSizeUnit(filesystem.min_size);
+      const partition: Partition = {
+        mountpoint: filesystem.mountpoint,
+        min_size: size.toString(),
+        unit: unit as Units,
+        id: uuidv4(),
+      };
+      return partition;
+    });
+
+    if (newPartitions) {
+      dispatch(changeFileSystemConfigurationType('manual'));
+      for (const partition of newPartitions) {
+        dispatch(addPartition(partition));
+      }
+    }
+  };
+
+  const handleServices = (services: Services | undefined) => {
+    dispatch(changeEnabledServices(services?.enabled || []));
+    dispatch(changeMaskedServices(services?.masked || []));
+    dispatch(changeDisabledServices(services?.disabled || []));
+  };
+
+  const handleKernelAppend = (kernelAppend: string | undefined) => {
+    dispatch(clearKernelAppend());
+
+    if (kernelAppend) {
+      const kernelArgsArray = kernelAppend.split(' ');
+      for (const arg of kernelArgsArray) {
+        dispatch(addKernelArg(arg));
+      }
+    }
+  };
+
+  const applyChanges = (selection: ComplianceSelectOptionValueType) => {
+    if (selection.policyID === undefined) {
+      // handle user has selected 'None' case
+      handleClear();
+    } else {
+      const oldOscapPackages = currentProfileData?.packages || [];
+      trigger(
+        {
+          distribution: release,
+          policy: selection.policyID,
+        },
+        true // preferCacheValue
+      )
+        .unwrap()
+        .then((response) => {
+          const oscapPartitions = response.filesystem || [];
+          const newOscapPackages = response.packages || [];
+          handlePartitions(oscapPartitions);
+          handlePackages(oldOscapPackages, newOscapPackages);
+          handleServices(response.services);
+          handleKernelAppend(response.kernel?.append);
+          const compl = selection as ComplianceSelectOptionValueType;
+          setSelected(compl.toString());
+          dispatch(
+            changeCompliance({
+              profileID: compl.profileID,
+              policyID: compl.policyID,
+              policyTitle: compl.toString(),
+            })
+          );
+        });
+    }
+  };
+
+  const handleSelect = (
+    _event: React.MouseEvent<Element, MouseEvent>,
+    selection: string
+  ) => {
+    if (selection) {
+      applyChanges(selection as unknown as ComplianceSelectOptionValueType);
+      setIsOpen(false);
+    }
+  };
+
+  const complianceOptions = () => {
+    if (!policies || policies.data === undefined) {
+      return [];
+    }
+
+    const res = [
+      <SelectOption
+        key="compliance-none-option"
+        value={{ toString: () => 'None', compareTo: () => false }}
```

Comment:
Why do we have this here? Is it not enough to not select any profile?

Reply:
The empty option is there to have a way to deselect a policy in case you already chose one. Similar to the ProfileSelector.

Reply:
But there is a way to do that now, using the close button. :thinking: I was just a bit confused, because we do not use this in different steps (like Timezone, Locale (Keyboard), etc.). But this is not a topic of this PR, so I am happy to just leave this as is, and open this conversation elsewhere. :)

https://github.com/user-attachments/assets/4deb4e18-cdc5-4c7e-b551-139660552574

Reply:
There is an inconsistency between policy and profile selectors - default value shown, rendering clear button - I'd leave it for a follow up as it's not entirely in the scope of this PR. But we should still address it at some point.